### PR TITLE
LK1571 - allow section toggling

### PR DESCRIPTION
**Pull request by crosiere18**
Allow user to pull up sections containing students when going into Attendance Summary from Class Roster with a blank section.

### CODE REVIEW

File: src/test/java/edu/ksu/canvas/attendance/repository/ReportRepositoryITest.java [hunk 1]
```diff
@@ -44,11 +41,15 @@
...
     private AttendanceCourseRepository courseRepository;
+
+    @Autowired
+    private AttendanceSectionRepository sectionRepository;
     
     @Autowired
     private ReportRepository reportRepository;
     
     private AttendanceCourse existingCourse;
+    private AttendanceSection existingSection;
```
Comment (KSU-Github) [inline src/test/java/edu/ksu/canvas/attendance/repository/ReportRepositoryITest.java]: MAJOR Remove this unused "existingSection" private field. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1068)

**Comment by KSU-Github** [comment]
SonarQube analysis reported 20 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 17 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [SummaryControllerITest.java#L109](https://github.com/kstateome/lti-attendance/blob/9a27fa8db483bf0f26e3a99fe99484b51728b55c/src/test/java/edu/ksu/canvas/attendance/controller/SummaryControllerITest.java#L109): Either log or rethrow this exception. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [ReportRepositoryITest.java#L64](https://github.com/kstateome/lti-attendance/blob/9a27fa8db483bf0f26e3a99fe99484b51728b55c/src/test/java/edu/ksu/canvas/attendance/repository/ReportRepositoryITest.java#L64): Cast one of the operands of this division operation to a "float". [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS2184)
1. ![MINOR][MINOR] [SummaryControllerITest.java#L108](https://github.com/kstateome/lti-attendance/blob/9a27fa8db483bf0f26e3a99fe99484b51728b55c/src/test/java/edu/ksu/canvas/attendance/controller/SummaryControllerITest.java#L108): Define a constant instead of duplicating this literal "/studentSummary/" 5 times. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [SummaryControllerITest.java#L152](https://github.com/kstateome/lti-attendance/blob/9a27fa8db483bf0f26e3a99fe99484b51728b55c/src/test/java/edu/ksu/canvas/attendance/controller/SummaryControllerITest.java#L152): Define a constant instead of duplicating this literal "sectionId" 4 times. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [SummaryControllerITest.java#L155](https://github.com/kstateome/lti-attendance/blob/9a27fa8db483bf0f26e3a99fe99484b51728b55c/src/test/java/edu/ksu/canvas/attendance/controller/SummaryControllerITest.java#L155): Define a constant instead of duplicating this literal "studentId" 4 times. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [SummaryControllerITest.java#L156](https://github.com/kstateome/lti-attendance/blob/9a27fa8db483bf0f26e3a99fe99484b51728b55c/src/test/java/edu/ksu/canvas/attendance/controller/SummaryControllerITest.java#L156): Define a constant instead of duplicating this literal "attendances" 4 times. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [SummaryControllerITest.java#L162](https://github.com/kstateome/lti-attendance/blob/9a27fa8db483bf0f26e3a99fe99484b51728b55c/src/test/java/edu/ksu/canvas/attendance/controller/SummaryControllerITest.java#L162): Define a constant instead of duplicating this literal "dateOfClass" 3 times. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [SummaryControllerITest.java#L177](https://github.com/kstateome/lti-attendance/blob/9a27fa8db483bf0f26e3a99fe99484b51728b55c/src/test/java/edu/ksu/canvas/attendance/controller/SummaryControllerITest.java#L177): Define a constant instead of duplicating this literal "entries" 3 times. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [ReportRepositoryITest.java#L57](https://github.com/kstateome/lti-attendance/blob/9a27fa8db483bf0f26e3a99fe99484b51728b55c/src/test/java/edu/ksu/canvas/attendance/repository/ReportRepositoryITest.java#L57): Make this final field static too. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1170)
1. ![MINOR][MINOR] [ReportRepositoryITest.java#L58](https://github.com/kstateome/lti-attendance/blob/9a27fa8db483bf0f26e3a99fe99484b51728b55c/src/test/java/edu/ksu/canvas/attendance/repository/ReportRepositoryITest.java#L58): Make this final field static too. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1170)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by kawesome** [comment]
Can one of the admins verify this patch?

File: src/test/java/edu/ksu/canvas/attendance/repository/ReportRepositoryITest.java (hunk 1 above)
Comment (japshvincent) [inline src/test/java/edu/ksu/canvas/attendance/repository/ReportRepositoryITest.java]: looks like this field was added and isn't used. I agree with sonar bot's suggestion.

**Pull request merged**